### PR TITLE
Remove scope validation in Access Token

### PR DESCRIPTION
**Pull request by vahancouver**
Change Access Token not to require scope. 
Update Unit Tests

### CODE REVIEW

File: FRAuth/FRAuth/OAuth2/AccessToken.swift [hunk 1]
```diff
@@ -20,7 +20,7 @@ import Foundation
     /// token_type of access_token
     @objc public var tokenType: String
     /// Granted scope(s) with space separator for given access_token
-    @objc public var scope: String
+    @objc public var scope: String?
```
Comment (george-bafaloukas-forgerock) [inline FRAuth/FRAuth/OAuth2/AccessToken.swift:23]: Maybe leave that as non-optional and remove the scope from the validation below. So the flow actually passes without breaking the public API. If you want what you can do is set this with an empty string before doing all the checks. Like:
let scope = tokenResponse[OAuth2.scope] as? String ?? ""

Comment (vahancouver) [reply FRAuth/FRAuth/OAuth2/AccessToken.swift]: @george-bafaloukas-forgerock made the change. Please have a look again

**Review by george-bafaloukas-forgerock — APPROVED** [review]

File: FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/AccessTokenTests.swift [hunk 2]
```diff
@@ -109,10 +109,11 @@ class AccessTokenTests: FRAuthBaseTest {
...
         // Given missing scope
-        tokenDict = ["access_token":self.access_token, "token_type":self.token_type, "expires_in":self.expires_in]
-        at = AccessToken(tokenResponse: tokenDict)
-        // Then
-        XCTAssertNil(at)
+        // Removing the validation for scope as it is no longer mandatory
+//        tokenDict = ["access_token":self.access_token, "token_type":self.token_type, "expires_in":self.expires_in]
+//        at = AccessToken(tokenResponse: tokenDict)
+//        // Then
+//        XCTAssertNil(at)
```
Comment (spetrov) [inline FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/AccessTokenTests.swift:116]: Probably better to leave the test and just change the assertion to `XCTAssertNotNil(at)`?

Comment (spetrov) [reply FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/AccessTokenTests.swift:116]: NM... this is covered in the `Given minimal set of response values` case...

File: FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/AccessTokenTests.swift [hunk 3]
```diff
@@ -140,9 +141,10 @@ class AccessTokenTests: FRAuthBaseTest {
         XCTAssertNil(at)
         
         // Given missing scope
-        at = AccessToken(token: self.access_token, expiresIn: self.expires_in, scope: nil, tokenType: self.token_type, refreshToken: nil, idToken: nil, authenticatedTimestamp: Date().timeIntervalSince1970)
-        // Then
-        XCTAssertNil(at)
+        // Removing the validation for scope as it is no longer mandatory
+//        at = AccessToken(token: self.access_token, expiresIn: self.expires_in, scope: nil, tokenType: self.token_type, refreshToken: nil, idToken: nil, authenticatedTimestamp: Date().timeIntervalSince1970)
+//        // Then
+//        XCTAssertNil(at)
```
Comment (spetrov) [inline FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/AccessTokenTests.swift:147]: May be better to change the assertion to `XCTAssertNotNil(at)`?

**Review by spetrov — APPROVED** [review]
Looks good to me. 👍🏻 
Have a few comments, but feel free to disregard them if you don't agree...

**Review by witrisna — APPROVED** [review]

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

**Pull request merged**